From 815ac17b571a007fe69e6d523b896dd6d4eb9772 Mon Sep 17 00:00:00 2001
From: haiqiang yao <haiqiangx.yao@intel.com>
Date: Fri, 4 Dec 2015 10:14:00 +0800
Subject: [PATCH 12/13] atomisp2:Add two new format ISP2.7

the patch will add two new format in atomisp drvier for
debugging ISP2.7.
IA_CSS_FRAME_FORMAT_YUV420_16
IA_CSS_FRAME_FORMAT_YCgCo444_16

Change-Id: I0bec51e8536b22e7b8ee9c7ba2b6ffe1df44bbeb
Original-Tracked-On: #H1604058903
Tracked-On:https://jira01.devtools.intel.com/browse/OAM-13009
Signed-off-by: haiqiang yao <haiqiangx.yao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/88128
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Build: Dessert M COHO (SC)
Build: Dessert M BXT RVP (SC)
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Reviewed-on: https://android.intel.com:443/450393
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c | 10 +++++++++-
 .../media/pci/atomisp2/atomisp_driver/atomisp_compat.h      |  1 +
 .../media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c       | 13 +++++++++++++
 .../media/pci/atomisp2/atomisp_driver/atomisp_subdev.c      |  3 +++
 4 files changed, 26 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 310ed68..086e918 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -1931,6 +1931,10 @@ v4l2_fmt_to_sh_fmt(u32 fmt)
 	case V4L2_PIX_FMT_JPEG:
 	case V4L2_PIX_FMT_CUSTOM_M10MO_RAW:
 		return CSS_FRAME_FORMAT_BINARY_8;
+	case V4L2_PIX_FMT_CUSTOM_YUV420_16:
+		return CSS_FRAME_FORMAT_YUV420_16;
+	case V4L2_PIX_FMT_CUSTOM_YCgCo444_16:
+		return CSS_FRAME_FORMAT_YCgCo444_16;
 	case V4L2_PIX_FMT_SBGGR16:
 	case V4L2_PIX_FMT_SBGGR10:
 	case V4L2_PIX_FMT_SGBRG10:
@@ -2004,6 +2008,8 @@ static u32 get_pixel_depth(u32 pixelformat)
 	case V4L2_PIX_FMT_SGBRG10:
 	case V4L2_PIX_FMT_SGRBG10:
 	case V4L2_PIX_FMT_SRGGB10:
+	case V4L2_PIX_FMT_CUSTOM_YUV420_16:
+	case V4L2_PIX_FMT_CUSTOM_YCgCo444_16:
 		return 16;
 	case V4L2_PIX_FMT_RGB24:
 	case V4L2_PIX_FMT_YUV444:
@@ -4402,6 +4408,9 @@ atomisp_bytesperline_to_padded_width(unsigned int bytesperline,
 		return bytesperline/2;
 	case CSS_FRAME_FORMAT_RGBA888:
 		return bytesperline/4;
+	case CSS_FRAME_FORMAT_YUV420_16:
+	case CSS_FRAME_FORMAT_YCgCo444_16:
+		return bytesperline/2;
 	/* The following cases could be removed, but we leave them
 	   in to document the formats that are included. */
 	case CSS_FRAME_FORMAT_NV11:
@@ -4412,7 +4421,6 @@ atomisp_bytesperline_to_padded_width(unsigned int bytesperline,
 	case CSS_FRAME_FORMAT_YV12:
 	case CSS_FRAME_FORMAT_YV16:
 	case CSS_FRAME_FORMAT_YUV420:
-	case CSS_FRAME_FORMAT_YUV420_16:
 	case CSS_FRAME_FORMAT_YUV422:
 	case CSS_FRAME_FORMAT_YUV422_16:
 	case CSS_FRAME_FORMAT_YUV444:
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat.h
index 5a404e4..2ecafa7 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat.h
@@ -120,6 +120,7 @@
 #define CSS_FRAME_FORMAT_RGBA888	CSS_ID(CSS_FRAME_FORMAT_RGBA888)
 #define CSS_FRAME_FORMAT_QPLANE6	CSS_ID(CSS_FRAME_FORMAT_QPLANE6)
 #define CSS_FRAME_FORMAT_BINARY_8	CSS_ID(CSS_FRAME_FORMAT_BINARY_8)
+#define CSS_FRAME_FORMAT_YCgCo444_16	CSS_ID(CSS_FRAME_FORMAT_YCgCo444_16)
 
 struct atomisp_device;
 struct atomisp_sub_device;
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index 8488eb9..89ad3b9 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -516,6 +516,19 @@ const struct atomisp_format_bridge atomisp_output_fmts[] = {
 		.mbus_code = V4L2_MBUS_FMT_CUSTOM_M10MO_RAW,
 		.sh_fmt = CSS_FRAME_FORMAT_BINARY_8,
 		.description = "Custom RAW for M10MO"
+	}, {
+	/* Only for ISP2.7 format*/
+		.pixelformat = V4L2_PIX_FMT_CUSTOM_YCgCo444_16,
+		.depth = 16,
+		.mbus_code = V4L2_MBUS_FMT_CUSTOM_YCgCo444_16,
+		.sh_fmt = CSS_FRAME_FORMAT_YCgCo444_16,
+		.description = "Custom YCgCo444_16"
+	}, {
+		.pixelformat = V4L2_PIX_FMT_CUSTOM_YUV420_16,
+		.depth = 16,
+		.mbus_code = V4L2_MBUS_FMT_CUSTOM_YUV420_16,
+		.sh_fmt = CSS_FRAME_FORMAT_YUV420_16,
+		.description = "Custom YUV420_16"
 	},
 };
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
index 14dee6a..af58232 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
@@ -56,6 +56,9 @@ const struct atomisp_in_fmt_conv atomisp_in_fmt_conv[] = {
 	{ V4L2_MBUS_FMT_CUSTOM_M10MO_RAW, 8, 8, CSS_FRAME_FORMAT_BINARY_8, 0, IA_CSS_STREAM_FORMAT_BINARY_8 },
 	/* no valid V4L2 MBUS code for metadata format, so leave it 0. */
 	{ 0, 0, 0, ATOMISP_INPUT_FORMAT_EMBEDDED, 0, IA_CSS_STREAM_FORMAT_EMBEDDED },
+	/* only for ISP2.7*/
+	{ V4L2_MBUS_FMT_CUSTOM_YCgCo444_16, 16, 16, CSS_FRAME_FORMAT_YCgCo444_16, 0, CSS_FRAME_FORMAT_YCgCo444_16 },
+	{ V4L2_MBUS_FMT_CUSTOM_YUV420_16, 16, 16, CSS_FRAME_FORMAT_YUV420_16, 0, CSS_FRAME_FORMAT_YUV420_16 },
 	{}
 };
 
-- 
1.9.1

